File: app/src/main/java/eu/szwiec/replayview/replay/FilesProvider.java
package eu.szwiec.replayview.replay;

import org.apache.commons.io.FilenameUtils;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

import eu.szwiec.replayview.FileUtils;
import eu.szwiec.replayview.SDKConstants;

public class FilesProvider {

    private FilesProvider() {
    }

    public static List<File> getFiles(String zipPath, CharSequence[] dataTypes) {
        FilesProvider provider = new FilesProvider();
        File extractedDir = provider.getExtractedDir(zipPath);
        List<File> files = provider.getMatchingFiles(dataTypes, extractedDir);

        return files;
    }

    private List<File> getMatchingFiles(CharSequence[] types, File dir) {
        List<File> files = new ArrayList<>();

        for(CharSequence type : types) {
            files.addAll( getMatchingFiles(type, dir) );
        }
        return files;
    }

    private List<File> getMatchingFiles(CharSequence type, File dir) {
        List<File> files = new ArrayList<>();

        String[] names = dir.list(
                (d, name) -> {
                    String regex = String.format(".*\\.%s\\d+", type);
                    return name.matches(regex);
                });

        for (String name : names) {
            File file = new File(dir + "/" + name);
            files.add(file);
        }

        return files;
    }

    private File getExtractedDir(String zipPath) {
        String destinationPath = getDestinationPath(zipPath);
        boolean isUnzipSuccessful = FileUtils.unzip(zipPath, destinationPath);

        if(!isUnzipSuccessful) throw new RuntimeException("Problem with unzipping");

        return new File(destinationPath);
    }

    private String getDestinationPath(String zipPath) {
        String filenameWithExtension = zipPath.substring(zipPath.lastIndexOf("/") + 1);
        return SDKConstants.SDCARD_PATH + FilenameUtils.removeExtension(filenameWithExtension);
    }
}

File: app/src/main/java/eu/szwiec/replayview/replay/FilesProvider.kt
package eu.szwiec.replayview.replay

import eu.szwiec.replayview.FileUtils
import eu.szwiec.replayview.SDKConstants
import org.apache.commons.io.FilenameUtils
import java.io.File
import java.util.*

object FilesProvider {

    fun getFiles(zipPath: String, extensions: List<String>): List<File> {
        val extractedDir = extractDir(zipPath)

        return getMatchingFiles(extensions, extractedDir)
    }

    private fun extractDir(zipPath: String): File {
        val destinationPath = getDestinationPath(zipPath)
        val isUnzipSuccessful = FileUtils.unzip(zipPath, destinationPath)

        if (!isUnzipSuccessful) throw RuntimeException("Problem with unzipping")

        return File(destinationPath)
    }

    private fun getDestinationPath(zipPath: String): String {
        val filenameWithExtension = zipPath.substring(zipPath.lastIndexOf("/") + 1)
        return SDKConstants.SDCARD_PATH + FilenameUtils.removeExtension(filenameWithExtension)
    }

    private fun getMatchingFiles(types: List<String>, dir: File): List<File> {
        val files = ArrayList<File>()

        for (type in types) {
            files.addAll(getMatchingFiles(type, dir))
        }
        return files
    }

    private fun getMatchingFiles(type: String, dir: File): List<File> {
        val files = ArrayList<File>()

        val names = dir.list { d, name ->
            val regex = String.format(".*\\.%s\\d+", type)
            name.matches(regex.toRegex())
        }

        for (name in names) {
            val file = File(dir.toString() + "/" + name)
            files.add(file)
        }

        return files
    }
}
